gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonIntermediateToParquetConverter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.parquet;
+
+import org.apache.gobblin.configuration.WorkUnitState;
+import org.apache.gobblin.converter.Converter;
+import org.apache.gobblin.converter.DataConversionException;
+import org.apache.gobblin.converter.SchemaConversionException;
+import org.apache.gobblin.converter.SingleRecordIterable;
+import org.apache.gobblin.converter.parquet.JsonElementConversionFactory.RecordConverter;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import parquet.example.data.Group;
+import parquet.schema.MessageType;
+
+import static org.apache.gobblin.converter.parquet.JsonElementConversionFactory.RecordConverter.RecordType.ROOT;
+
+
+/**
+ * A converter to Convert JsonIntermediate to Parquet
+ * @author tilakpatidar
+ */
+public class JsonIntermediateToParquetConverter extends Converter<JsonArray, MessageType, JsonObject, Group> {
```

Comment:
1. This isn't really a general Parquet converter. Because it only works for `Group` and doesn't work for `Primitive`. How about naming it as `JsonToParquetGroupConverter`?

2. `MessageType` -> `GroupType`?

Reply:
Please see the following links.
MessageType is a root of the schema. [MessageType.java](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/schema/MessageType.java)
GroupType is a list of fields. [GroupType.java](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/schema/GroupType.java)

Data from any source will have RECORD type as the root, thus converting a RECORD type data to MessageType data makes sense. This MessageType will act as a parent for other GroupType which are used to build the MAP, ARRAY etc.

So, MessageType is the root schema. 

We don't need a JsonParquetPrimitiveConverter as the source data will always be a RECORD.
However, I agree that renaming to ```JsonToParquetGroupConverter``` will clarify the usage.